### PR TITLE
implements hit results with filtering on Sqlite/geopackage dataprovier

### PR DESCRIPTION
**Pull request by jorgejesus**
Implements sqlite3/gpkg filtering on hits just like posgresql

- private method `__get_where_clauses` to generate where clauses
- removal of pytest that where redundant, since they would call same code for sqlite/gpkg

### CODE REVIEW

File: pygeoapi/provider/sqlite.py [hunk 1]
```diff
@@ -153,7 +186,8 @@ def __load(self):
         # conn.set_trace_callback(LOGGER.debug)
         cursor = conn.cursor()
         try:
-            cursor.execute(f"SELECT load_extension('{SPATIALITE_EXTENSION}')")
```
[the remark this answers is not in the document]
Comment (tomkralidis) [reply pygeoapi/provider/sqlite.py]: This is rooted in my recommendations on using `.format()`. Given Travis is only rooted in 3.x, no opposition from me to use f-strings then.  We should (at a later point) advise on our common dev idioms in our documentation.

[the remark this answers is not in the document]
Comment (francbartoli) [reply pygeoapi/provider/sqlite.py]: 👍

[the remark this answers is not in the document]
Comment (jorgejesus) [reply pygeoapi/provider/sqlite.py]: But for consistency I will change it , forgot about it

[the remark this answers is not in the document]
Comment (jorgejesus) [reply pygeoapi/provider/sqlite.py]: This seems to have been correct by latest commit: https://github.com/geopython/pygeoapi/commit/72b9a647ff2f9e866a7abc2334714a3a87c74096

File: pygeoapi/provider/sqlite.py [hunk 2]
```diff
@@ -95,6 +95,39 @@ def get_fields(self):
...
+        Generarates WHERE conditions to be implemented in query.
+        Private method mainly associated with query method.
+
+        Method returns part of the SQL query, plus tupple to be used
+        in the sqlite query method
+
+        :param properties: list of tuples (name, value)
+        :param bbox: bounding box [minx,miny,maxx,maxy]
+
+        :returns: str, tupple
```
[the remark this answers is not in the document]
Comment (jorgejesus) [reply pygeoapi/provider/sqlite.py]: Fixed

**Comment by jorgejesus** [comment]
- Synced fork code with geopython:master (#359)
- Fixed review requests from @francbartoli 
- SQL queries with caps key words

**Pull request merged**